apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -725,6 +780,12 @@ export class WorkshopForm extends React.Component {
 
   handleCourseChange = event => {
     const course = this.handleFieldChange(event);
+    const cannotChangeIfWorkshopVirtual =
+      this.checkCannotChangeIfWorkshopVirtual(
+        this.state.sessions,
+        course,
+        this.state.subject
+      );
```

Comment:
heads up that this is leading to some weird behavior: 

https://user-images.githubusercontent.com/8001765/234135048-cd2cdf09-dcf4-4bce-b507-5bdca51b38c6.mov

you're using the new course value but the old subject value to compute the result. in this case it is probably harmless, because the problem is fixed after you select a subject, but please keep this kind of issue in mind for the future.

better alternatives for next time include:
* avoid duplicate state entirely by calling checkCannotChangeIfWorkshopVirtual only when you need it
* set a constant equal to the result of checkCannotChangeIfWorkshopVirtual at the beginning of the render method, without assigning it to component state
* if you must use state, consider using [setState callbacks](https://legacy.reactjs.org/docs/react-component.html#setstate) to ensure that all state updates have finished before relying on this.state (unsure what the equivalent pattern is using react hooks)

Reply:
Oh good point! That change was in the case that 'course' was being changed so that's why I used the newly-defined 'course' but the state-defined 'subject' and 'sessions'. You're right, there is no need for cannotChangeIfWorkshopVirtual in the state so I'll just remove it to keep things simpler and only call it when I need it! (I double checked after removing it from state and the behavior is the same and much less complicated!)